### PR TITLE
Fix: During refresh, rhs flickers with wrong message 

### PR DESCRIPTION
**Pull request by tintinthong**
https://linear.app/cardstack/issue/CS-6157/[bug]-while-loading-schema-editor-shows-the-wrong-message

## Notice there is no ficker anymore

Handled for the card json and module case

https://github.com/cardstack/boxel/assets/8165111/42f77de2-ec49-40e4-98a6-73684a1248ff

## File Error
Screenshot 2023-11-03 at 6 24 58 PM

## No card no field Error 
Screenshot 2023-11-03 at 6 25 24 PM

## Selection not a card or field Error 

I brought this up yesterday, it is arguable that we would show this error with schema editor accordion item. But I concluded to NOT show it by principle that the schema editor will only show cards or fields. So its actually original behaviour.

Screenshot 2023-11-03 at 6 25 52 PM

## Additional refactors 

https://linear.app/cardstack/issue/CS-6207/refactoring-code-submodegts

- Rename schemaEditorIncompatibleFile to isFileIncompatibleWithSchemaEditor 
  - **Reduced to rhsIncompatibleMessage. We are saying that nothing on rhs is compatible.**
- Rename schemaEditorIncompatibleItem to isSelectedItemIncompatibleWithSchemaEditor 
  - **Reduced to rhsIncompatibleMessage. We are saying that nothing on rhs is compatible. **
- Rename isValidSchemaFile to hasCardDefOrFieldDef 
  - **Done**
- Consider moving schema editor incompat messages into SchemaEditorColumn 
  - **Nope still in code mode. All the messages were really about non cards or fields. Nothing the schema editor would show**
- moved module resource into module contents resource (since this is the only depedent). 
  - **This was not in the linear for refactoring**

### CODE REVIEW

**Comment by tintinthong** [comment]
Ok pause on review. Will add changes based upon feedback in boxel. I will scope to see if pulling accordion out is doable in this pr.

**Comment by jurgenwerk** [comment]
Currently, when you open a module in the url bar, the schema editor will show "Schema Editor cannot be used with this file type" for a split second before it will show the schema editor. This is a bit unexpected and would be great to fix - it looks like this PR is touching this area so just bringing this issue to awareness. It is more noticeable in slow loading environments such as staging.

**Comment by tintinthong** [comment]
> Currently, when you open a module in the url bar, the schema editor will show "Schema Editor cannot be used with this file type" for a split second before it will show the schema editor. This is a bit unexpected and would be great to fix - it looks like this PR is touching this area so just bringing this issue to awareness. It is more noticeable in slow loading environments such as staging.

yup noticed it. should be fixed in this pr.

**Comment by tintinthong** [comment]
This is ready for review. @lukemelia I paraphrased you text and used it to work with the error logic that was pre-existing. 

Decided not to carry out the accordion refactor in this PR -- but I think it should be much easier to implement now

File: packages/host/app/components/operator-mode/code-submode.gts [hunk 1]
```diff
@@ -916,32 +954,7 @@ export default class CodeSubmode extends Component<Signature> {
...
-                      Schema Editor cannot be used for selected
-                      {{this.selectedDeclaration.type}}
-                      "{{this.selectedDeclaration.localName}}".</div>
-                  {{else if this.cardError}}
-                    {{this.cardError.message}}
                   {{/if}}
-                {{else if this.isLoading}}
-                  <div class='loading'>
-                    <LoadingIndicator />
-                  </div>
```
Comment (tintinthong) [inline packages/host/app/components/operator-mode/code-submode.gts:944]: These conditional templates were kind of confusing. I realise that all the errors shown actually just mean that NOTHING on the rhs can show it. So I coupled everything into `rhsIncompatibleMessage`

File: packages/host/app/components/operator-mode/code-submode.gts [hunk 2]
```diff
@@ -252,6 +256,37 @@ export default class CodeSubmode extends Component<Signature> {
     return !this.codePath || this.currentOpenFile?.state === 'not-found';
   }
 
+  private get rhsIncompatibleMessage() {
```
Comment (jurgenwerk) [inline packages/host/app/components/operator-mode/code-submode.gts]: can we choose a more descriptive and robust naming than "rhs*", for example "fileIncompatibilityMessage"?

**Review by burieberry — APPROVED** [review]

**Review by jurgenwerk — APPROVED** [review]
It looks like it solves the issues but I would stay away from "rhs" naming. I suggest to name it so that it is not influenced by its placement in the layout.

**Pull request merged**